### PR TITLE
Bespin Production Deployment

### PR DESCRIPTION
**Pull request by johnbradley**
Deploys Bespin via two steps:
- Run openstack heat template
- Run ansible inside deploy VM created via heat template

The database, webserver and lando run inside docker containers inside of openstack VMs.
The lando_worker, rabbit and nfs servers are deployed without docker.

### CODE REVIEW

File: bespin_heat/bespin-infrastructure.heat [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  ssh_security:
+    type: OS::Neutron::SecurityGroup
+    properties:
+      name: ssh_access
+      description: SSH
+      rules:
+      - protocol: tcp
+        port_range_min: 22
+        port_range_max: 22
+  postgres_security:
```
Comment (dleehr) [inline bespin_heat/bespin-infrastructure.heat]: I don't think we need all these security groups - they're for traffic that crosses into the private network from outside it (e.g. the internet or other networks).

We'd only need security groups for things we need to connect to from outside, like ssh on the deploy host and the web http/https ports.

Comment (dleehr) [inline bespin_heat/bespin-infrastructure.heat]: Or in other words, things with floating IPs

File: roles/bespin_rabbit/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+- name: Install v3.6.6 rabbitmq from their site since standard repositories rabbitmq-server is really out of date
```
Comment (dleehr) [inline roles/bespin_rabbit/tasks/main.yaml]: Trying to remember why this one service wasn't dockerized. Was it awkward to create users?

Comment (johnbradley) [inline roles/bespin_rabbit/tasks/main.yaml]: The docker container only lets you create a single user before starting. I believe this is part of the rabbitmq code and not something happening in the container.
With rabbitmq you have to launch the service before you can add a user.
So this breaks the wait until a port is ready docker script.
The ansible module knows to wait until the user is added before continuing.

**Comment by johnbradley** [comment]
I should probably add the ubuntu user to the docker group so the workflow can be run as a regular user.

File: roles/bespin_rabbit/tasks/main.yaml (hunk 2 above)
Comment (dleehr) [inline roles/bespin_rabbit/tasks/main.yaml]: Yeah I thought it was something like that. A quick googling shows we'd likely need to build our own rabbitmq docker image to dockerize this. One way is to add an entrypoint script that creates the users specified, then runs the requested docker command. Probably out of scope for this PR though.

**Comment by dleehr** [comment]
> I should probably add the ubuntu user to the docker group so the workflow can be run as a regular user.

On the lando worker host?

**Comment by johnbradley** [comment]
> On the lando worker host?
Yep. Right now we are running as root.

File: bespin_heat/bespin-infrastructure.heat (hunk 1 above)
Comment (johnbradley) [inline bespin_heat/bespin-infrastructure.heat]: On the production openstack I am unable to ssh into the nodes without this explicit security.
I think this is due to having a private network with `Shared: No`.

Comment (johnbradley) [inline bespin_heat/bespin-infrastructure.heat]: I am not seeing that behavior on the production openstack. If I don't enable ssh security I can't get from the deploy node to any of the others.

Comment (johnbradley) [inline bespin_heat/bespin-infrastructure.heat]: Looks like this was due to not having the default security group in instances created by heat.

Comment (dleehr) [inline bespin_heat/bespin-infrastructure.heat]: Yep, the confusion was that instances created through horizon get the security group `default` by default, which allows connections from other hosts in the `default` group. So my assumption was that this group was applied to everything.

**Pull request merged**